src/main/scala/analysis/MemoryModelMap.scala

```diff
@@ -1,32 +1,49 @@
 package analysis
 
 import analysis.*
-import ir.BitVecLiteral
+import ir.{BitVecLiteral, Procedure}
 
 import scala.collection.mutable
 
 // Define a case class to represent a range
-case class RangeKey(var start: BigInt, var end: BigInt):
+case class RangeKey(var start: BigInt, var end: BigInt) extends Ordered[RangeKey]:
   def size(): BigInt = end - start
+  override def compare(that: RangeKey): Int = {
+    if (start < that.start) -1
+    else if (start > that.start) 1
+    else 0
+  }
 
 case class RegionToRangesMap():
-  val stackMap: mutable.Map[RangeKey, StackRegion] = mutable.Map()
-  val heapMap: mutable.Map[RangeKey, HeapRegion] = mutable.Map()
-  val dataMap: mutable.Map[RangeKey, DataRegion] = mutable.Map()
+  val stackMap: mutable.Map[RangeKey, StackRegion] = mutable.TreeMap()
+  val sharedStackMap: mutable.Map[Procedure, mutable.TreeMap[RangeKey, StackRegion]] = mutable.Map[Procedure, mutable.TreeMap[RangeKey, StackRegion]]()
+  val heapMap: mutable.Map[RangeKey, HeapRegion] = mutable.TreeMap()
+  val dataMap: mutable.Map[RangeKey, DataRegion] = mutable.TreeMap()
 
```

Comment:
Why is this a case class and why is it even separate from the MemoryModelMap class? these could all just be internal variables to MemoryModelMap?